IsIdentifiable/Failures/Failure.cs

```diff
@@ -12,41 +11,41 @@ namespace IsIdentifiable.Failures;
 /// along with the <see cref="ResourcePrimaryKey"/> (if any) so that the row on which
 /// the data was found can located (e.g. to perform  redaction).
 /// </summary>
-public class Failure : MemberwiseEquatable<Failure>
+public sealed record Failure
 {
     /// <summary>
     /// Each sub part of <see cref="ProblemValue"/> that the system had a problem with
     /// </summary>
-    public ReadOnlyCollection<FailurePart> Parts { get; private set; }
+    public ReadOnlyCollection<FailurePart> Parts { get; }
 
     /// <summary>
     /// Description of the item being evaluated (e.g. table name, file name etc)
     /// </summary>
-    public string Resource { get; set; }
+    public string? Resource { get; init; }
```

Comment:
This seems unrelated to the Equ removal? Also, I don't think this (and others) should ever be null really.

Reply:
Have broken some bits out into #573, will refine and resubmit the Equ bits after that